skipchain/skipchain.go

```diff
@@ -343,16 +343,12 @@ func (s *Service) StoreSkipBlock(psbd *StoreSkipBlock) (*StoreSkipBlockReply, er
 					"Didn't get skipblock in back-link")
 
 			}
-			if err := s.forwardLink(
-				&ForwardSignature{
-					TargetHeight: i + 1,
-					Previous:     back.Hash,
-					Newest:       prop,
-				}); err != nil {
-				// This is not a critical failure - we have at least
-				// one forward-link
-				log.Lvl1("Couldn't get old block to sign: " + err.Error())
-			}
+			// Requesting creation of secondary forward link.
+			s.SendRaw(back.Roster.List[0], &ForwardSignature{
```

Comment:
So this is a "fire and forget" operation, right? No feedback if it fails? Do we have a test that shows what happens if the leader is down and misses adding a forward signature?

Reply:
It's "fire and forget", because the secondary forward-links are only needed to traverse faster.
Now that I think of it, I need to add something to correctly add forward-links if previous ones failed :( R4M -> WIP...